tests/callbacks/test_EnqueuedEventTimer.cpp

```diff
@@ -49,17 +50,55 @@ TEST_F(EnqueuedEventTimerTest, duration)
 
     this->timer.stop(EndParallelForEvent{.event_id = 1});
 
-    const double elapsed = timer.template duration<Kokkos::utils::timer::milliseconds>().count();
+    const double elapsed = this->timer.template duration<Kokkos::utils::timer::milliseconds>().count();
 
-    //! Check that the elapsed time is greater than the waiting time of 100 ms.
-    ASSERT_GE(elapsed, 100.);
+    //! Check that the elapsed time is greater than 0.
+    ASSERT_GE(elapsed, 0.);
 }
 
+struct EnqueuedEventWithLaunchTimerTest : public TimerTest<EnqueuedEventWithLaunchTimer<execution_space>> {};
+
+//! @test Check the duration and launch reported by @ref Kokkos::utils::callbacks::EnqueuedEventWithLaunchTimer.
+TEST_F(EnqueuedEventWithLaunchTimerTest, duration)
+{
+    Kokkos::utils::timer::Timer<void> timer_external {};
+
+    timer_external.start();
+
+    this->timer.start(BeginParallelForEvent{.name = "my-name", .dev_id = this->dev_id, .event_id = 1});
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
```suggestion
    std::this_thread::sleep_for(std::chrono::microseconds(5));
```

Let's get closer to some realistic launch overhead.

Reply:
OK. More relevant indeed!